src/main/kotlin/uk/co/craigbass/pratura/usecase/ViewAllProducts.kt

```diff
@@ -0,0 +1,15 @@
+package uk.co.craigbass.pratura.usecase
+
+import uk.co.craigbass.pratura.boundary.ViewAllProducts
+import uk.co.craigbass.pratura.boundary.ViewAllProducts.PresentableProduct
+import java.math.BigDecimal
+
+class ViewAllProducts(val productRetriever: ProductRetriever) : ViewAllProducts {
+  override fun execute(request: Unit): List<PresentableProduct> {
+    return productRetriever.all().map { product ->
+      PresentableProduct(product.name, "£${product.price.toCurrencyString()}", product.sku)
```

Comment:
Yeah Unit largely means Nothing in Kotlin. it's because all concrete "SynchronousUseCase" implementations need a request param... but here we have nothing in our request.